python/beeai_framework/context.py

```diff
@@ -82,7 +83,7 @@ async def _run_tasks(self) -> R:
 
         return await self.handler()
 
-    def _set_context(self, context: dict) -> None:
+    def _set_context(self, context: dict[str, Any]) -> None:
```

Comment:
not sure if the implementation is correct, we might want to merge stuff instead of replacing them